api/server.js

```diff
@@ -13,52 +16,103 @@ app.get('/', (req, res) => {
 });
 
 app.get('/api/foods', (req, res) => {
-  res.send(app.locals.foods)
+  database.raw('SELECT * FROM foods').then(data => res.json(data.rows))
+})
+
+app.get('/api/foods/:id', (req, res) => {
+  const id = req.params.id
+  showFood(id, res)
 })
 
 app.post('/api/foods', (req, res) => {
   const calories = req.body.calories
   const name = req.body.name
-  if (app.locals.foods[name]) { return res.status(422).send(`${name} already exists!`)}
-  if (name && calories) {
-    app.locals.foods[name] = calories
-    return  res.status(201).json({name, calories})
-  }
-  return res.status(422).send({ error: 'Needs name and calories' })
+  if (name && calories) { return createFood(name, calories, res) }
+  res.status(422).send({ error: 'Needs name and calories' })
 })
 
-app.get('/api/foods/:name', (req, res) => {
-  const name = req.params.name
-  const calories = app.locals.foods[name]
-  if (!calories) { return res.sendStatus(404) }
-  res.json({name, calories})
+app.put('/api/foods/:id', (req, res) => {
+  const id = req.params.id
+  const calories = req.body.calories 
+  const name = req.body.name
+  updateFood(id, name, calories, res)
 })
 
-app.put('/api/foods/:name', (req, res) => {
-  const oldName = req.params.name
-  const calories = req.body.calories
-  const newName = req.body.name
-  if (calories && app.locals.foods[oldName]) {
-    if (oldName != newName) { delete app.locals.foods[oldName] }
-    app.locals.foods[newName] = calories
-    return  res.json({newName, calories})
-  }
-  return res.status(404).send({ error: `Could not find food ${oldName}` })
+app.delete('/api/foods/:id', (req, res) => {
+  const id = req.params.id
+  deleteFood(id, res)
 })
 
-app.delete('/api/foods/:name', (req, res) => {
-  const name = req.params.name
-  if (app.locals.foods[name]) { 
-    delete app.locals.foods[name] 
-    return res.status(200).send(`${name} deleted.`)
-  }
-  return res.status(404).send({ error: `Could not find food ${name}` })
-})
+function respondWith404(res, id) {
+  res.status(404).send({ error: `Could not find food with id ${id}` })
+}
+
+function respond(data, id, res) {
+  if (!data.rowCount) { return res.sendStatus(404) }
+  res.json(data.rows[0])
+}
+
+function showFood(id, res) {
+  database.raw('SELECT * FROM foods WHERE id = ?', id)
+  .then(data => respond(data, id, res))
+}
+
+function createFood(name, calories, res) {
+  database.raw(`INSERT INTO foods
+    (name, calories, created_at)
+    VALUES (?, ?, ?)
+    RETURNING id`,
+    [name, calories, new Date])
+  .then(data => database.raw('SELECT * FROM foods WHERE id = ?', data.rows[0].id))
+  .then(data => res.status(201).json(data))
+}
+
+function updateFood(id, name, calories, res) {
+  if (name && calories) { return updateNameCalories(id, name, calories, res) }
+  if (calories) { return updateCalories(id, calories, res) }
+  if (name) { return updateName(id, name, res) }
+}
```

Comment:
I think this was definitely the path of least resistance and allows us a clean way to figure it out differently later. I really like this, it reads well.. especially with the issues we had in pairin' alley.